client/CPlayerInterface.cpp

```diff
@@ -198,48 +199,200 @@ void CPlayerInterface::playerStartsTurn(PlayerColor player)
+	std::string mapDescription = cb->getMapHeader()->description;
+	std::string campaignName = "NA";
+	std::string timeStamp = cb->getStartInfo()->startTimeIso8601; // ToDo: krs - Get date time in more human format
+	std::string turn = std::to_string(cb->getDate(Date::MONTH))  
+		+ std::to_string(cb->getDate(Date::WEEK))
+		+ std::to_string(cb->getDate(Date::DAY_OF_WEEK)); 
+
+	// get random map info
+	if(mapName == "Random Map")
 	{
-		bool usePrefix = settings["general"]["useSavePrefix"].Bool();
-		std::string prefix = std::string();
+		mapType = RandomMap;
+		mapName = cb->getStartInfo()->mapGenOptions->getMapTemplate()->getName(); // template name
+		//auto playerCount = cb->getStartInfo()->mapGenOptions->getPlayerCount();
+	}
 
-		if(usePrefix)
-		{
-			prefix = settings["general"]["savePrefix"].String();
-			if(prefix.empty())
-			{
-				std::string name = cb->getMapHeader()->name;
-				int txtlen = TextOperations::getUnicodeCharactersCount(name);
+	// get campaign info
+	if(cb->getStartInfo()->mode == StartInfo::CAMPAIGN)
+	{
+		mapType = Campaign;
+		campaignName = cb->getStartInfo()->getCampaignName();
+	}
 
-				TextOperations::trimRightUnicode(name, std::max(0, txtlen - 15));
-				std::string forbiddenChars("\\/:?\"<>| ");
-				std::replace_if(name.begin(), name.end(), [&](char c) { return std::string::npos != forbiddenChars.find(c); }, '_' );
+	// get human player names (, separated) and number
+	std::string playerNames;
+	for(PlayerColor player(0); player < PlayerColor::PLAYER_LIMIT; ++player)
+	{
+		if(!cb->getStartInfo()->playerInfos.count(player))
+			continue;
 
-				prefix = name + "_" + cb->getStartInfo()->startTimeIso8601 + "/";
-			}
+		auto playerInfo = cb->getStartInfo()->playerInfos.at(player);
+		if(playerInfo.isControlledByHuman())
+		{
+			playerNames += playerInfo.name + ", ";
+			humanPlayerCount++;
 		}
+	}
+	playerNames.pop_back();
+	playerNames.pop_back();
+	isMulitplayerGame = humanPlayerCount > 1;
+
+	// normalize map name
+	//int txtlen = TextOperations::getUnicodeCharactersCount(mapName); 
+	//TextOperations::trimRightUnicode(mapName, std::max(0, txtlen - 15)); // shorten to max 15 chars
+	std::string forbiddenChars("\\/<>:\"|?*\t\n\v\f\r"); 
+	std::replace_if(mapName.begin(), mapName.end(), [&](char c) { return std::string::npos != forbiddenChars.find(c); }, '_' ); // replace forbidden chars with _
+
+	// store variable values in map
+	std::unordered_map<std::string, std::string> variables;
+	variables["mapName"] = mapName;
+	variables["templateName"] = mapName;
+	variables["timestamp"] = timeStamp;
+	variables["turn"] = turn;
+	variables["players"] = playerNames;
+	variables["campaignName"] = campaignName;
+ 
+	if(simpleSaves)
+	{
+		int autosaveCountLimit = settings["general"]["autosaveCountLimit"].Integer(); 
+		// save using a counter, and overwrite depending on autosave count limit
+		if(autosaveCountLimit > 0) 
+		{
+			auto saveNumber = extractNumberAfterHash(lastMapSaveName);

```

Comment:
Wouldn't it better and cleaner to save this information in the save file itself and get it from there? Then you can also make the counter as variable.

Reply:
will look how to make that.

Reply:
I think save file should not store this information. For now this will work.
Maybe in the future we get rid of it altogether? Or change how it works if Quick Load / Quick Save is implemented?